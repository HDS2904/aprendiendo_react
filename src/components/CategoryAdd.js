import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const CategoryAdd = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value )
    }

    const handleSubmit = (e) => {
        e.preventDefault(); // Elimina la acciÃ³n por defecto del tag form (envio del evento por default)
        if ( inputValue.trim().length > 1){
            setCategories( cats => [ inputValue, ...cats ]) // Permite agregar nuevo valor sin necesidad de mandar categorias del otro componente
            setInputValue('')   // Limpia el input
        }
    }

    return (
        <form onSubmit={handleSubmit}>
          <p>{inputValue}</p>
            <input 
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>
    )
}

// Propiedades del componente
CategoryAdd.protoType = {
    setCategories:  PropTypes.func.isRequired
}