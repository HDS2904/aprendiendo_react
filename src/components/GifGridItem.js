import React from 'react'


export const GifGridItem = ({title,url}) => {
    return (
        <div className="card animate__animated animate__backInRight">
            <img className="card__img" src={url} alt={title}/>
            <p className="card__title">{title}</p>
        </div>
    )
}
