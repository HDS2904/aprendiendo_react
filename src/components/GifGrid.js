import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async () => {
        const URL = 'https://api.giphy.com/v1/gifs/search?q=one+piece&limit=10&api_key=uGVVa23uqkXVnyco6Ztdkc5Xut4ZpaCL';
        const resp = await fetch(URL);
        const {data} = await resp.json();
        const gifs = data.map( item => {
            return {
                id: item.id,
                title: item.title,
                url: item.images?.downsized_medium.url,
            }
        })
        setImages(gifs);
    }


    return (
        <li>
            <h2>{category}</h2>
            <ol>
                {
                    images.map( ( {id, ...image} ) => {
                        return <GifGridItem 
                            key={id}
                            {...image}
                        />
                    })
                }
            </ol>
        </li>
    )
}
