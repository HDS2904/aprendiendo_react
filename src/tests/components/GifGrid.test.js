import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"
jest.mock("../../hooks/useFetchGifs")

describe('Prueba del componente GifGrid', () => {

  const category = 'One piece'
  
  test('EvaluaciÃ³n del contenido del componente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    })
    const wrapper = shallow( <GifGrid category={category} /> )
    expect( wrapper ).toMatchSnapshot()
  })
  
  test('Retorna los items cuando se carga las imagenes de useFetchGifs', () => {
    const gifs = [{
      id: 'ABC',
      url: 'hhtp://jonathan-hds.com.pe',
      title: 'lo que sea'
    }]
    useFetchGifs.mockReturnValue({
      loading: false,
      data: gifs
    })
    const wrapper = shallow( <GifGrid category={category} /> )
    expect( wrapper ).toMatchSnapshot()

    
  })
  

})
