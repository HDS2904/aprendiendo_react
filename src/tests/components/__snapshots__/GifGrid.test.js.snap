// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente GifGrid Evaluaci√≥n del contenido del componente 1`] = `
<Fragment>
  <h3>
    One piece
  </h3>
  <p>
    LOADING...
  </p>
  <div
    className="container"
  />
</Fragment>
`;
